docs/en/sql-reference/functions/string-functions.md

```diff
@@ -1196,3 +1196,53 @@ Result:
 Same as concatWithSeparator, the difference is that you need to ensure that concatWithSeparator(sep, expr1, expr2, expr3...) → result is injective, it will be used for optimization of GROUP BY.
 
 The function is named “injective” if it always returns different result for different values of arguments. In other words: different arguments never yield identical result.
+
+## extractKeyValuePairs
+
+Extracts key value pairs from any string. The string does not need to be 100% structured in a key value pair format,
+it might contain noise (e.g. log files). The key value pair format to be interpreted should be specified via function arguments.
+A key value pair consists of a key followed by a key_value_pair_delimiter and a value. 
+Special characters (e.g. $!@#¨) must be escaped. Enclosed/ quoted keys and values are accepted.
+
+The below grammar is a simplified representation of what is expected/ supported (does not include escaping and character allow_listing):
+
+* line = (reserved_char* key_value_pair)*  reserved_char*
+* key_value_pair = key kv_separator value
+* key = <quoted_string> |  asciichar asciialphanumeric*
+* kv_separator = ':'
+* value = <quoted_string> | asciialphanum*
```

Comment:
I don't see <quoted_string> explained in that grammar

I don't see value_special_characters_allow_list used in that grammar 

I don't see escape_character used in that grammar